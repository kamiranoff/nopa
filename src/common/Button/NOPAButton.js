import React, { PropTypes } from 'react';
import { TouchableHighlight, View, Dimensions } from 'react-native';

import { NOPAText } from '../index';

const NOPAButton = ({
                      fontFamily,
                      color,
                      textStyle,
                      onPress,
                      title,
                      backgroundColor,
                      children,
                      isDisabled,
                    }) => (
  <TouchableHighlight
    title={title}
    color={color}
    backgroundColor={backgroundColor}
    style={[{
      backgroundColor
    }, textStyle]}
    onPress={onPress}
    disabled={isDisabled}
  >
    <View>
      <NOPAText
        textStyle={{
          padding: 20,
          textAlign: 'center',
          color: '#fff'
        }}
      >
        {title}
      </NOPAText>
    </View>
  </TouchableHighlight>
);

NOPAButton.defaultProps = {
  textStyle: {
    marginTop: 10,
    marginBottom: 5,
    width: Dimensions.get('window').width - (Dimensions.get('window').width / 10),

  },
  backgroundColor: '#F97056'
};

NOPAButton.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  color: PropTypes.string,
  textStyle: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.object,
  ]),
  backgroundColor: PropTypes.string,
};

export default NOPAButton;
