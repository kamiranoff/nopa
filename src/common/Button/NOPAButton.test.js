import React from 'react';
import renderer from 'react-test-renderer';
import NOPAButton from './NOPAButton';

describe('<NOPAButton />', () => {
  it('Should match the snapshot', () => {
    const tree = renderer.create(
      <NOPAButton
        title={'I am a button'}
        onPress={() => console.log('Pressed')}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});