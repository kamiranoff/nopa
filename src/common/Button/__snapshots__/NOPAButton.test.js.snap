// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NOPAButton /> Should match the snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "backgroundColor": "#F97056",
        },
        Object {
          "marginBottom": 5,
          "marginTop": 10,
          "width": 675,
        },
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#444",
            "fontFamily": "monserratSemiBold",
            "fontSize": 18,
          },
          Object {
            "color": "#fff",
            "padding": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      I am a button
    </Text>
  </View>
</View>
`;
